Source/Particles/Pusher/GetAndSetPosition.H

```diff
@@ -15,6 +15,29 @@
 
 #include <limits>
 
+/** \brief Extract the cartesian position coordinates of the particle
+ *         p and store them in the variables `x`, `y`, `z`
+ *         This is the implementaton and can't be called directly */
+template <typename P>
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void getParticlePosition (const P& p, const RType theta,
+                          RType& x, RType& y, RType& z) const noexcept
+{
+#ifdef WARPX_DIM_RZ
+    RType r = p.pos(0);
+    x = r*std::cos(theta);
+    y = r*std::sin(theta);
+    z = p.pos(1);
+#elif WARPX_DIM_3D
+    x = p.pos(0);
+    y = p.pos(1);
+    z = p.pos(2);
+#else
+    x = p.pos(0);
+    y = m_snan;
```

Comment:
```suggestion
    y = std::numeric_limits<RType>::quiet_NaN();
```

Reply:
Not sure if we can use that directly in device code - let me test.